File: generate-inventory.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Generador automático de inventario de fotos
Escanea las carpetas de fotografías y genera inventario-auto.js
Uso: python3 generate-inventory.py
"""

import os
import json
import re
from pathlib import Path

def obtener_archivos_fotos(ruta_carpeta):
    """Obtiene lista de archivos de foto en una carpeta"""
    extensiones_validas = {'.jpg', '.jpeg', '.png', '.gif', '.webp'}
    
    try:
        archivos = os.listdir(ruta_carpeta)
        fotos = sorted([
            f for f in archivos 
            if os.path.isfile(os.path.join(ruta_carpeta, f)) 
            and os.path.splitext(f)[1].lower() in extensiones_validas
        ], key=lambda x: [int(n) if n.isdigit() else n for n in re.findall(r'\d+|\D+', x)])
        return fotos
    except Exception as e:
        print(f"⚠️ Error leyendo {ruta_carpeta}: {e}")
        return []

def generar_inventario():
    """Genera el inventario automático"""
    ruta_fotos = Path(__file__).parent / 'fotografias'
    categorias_activas = []
    
    if not ruta_fotos.exists():
        print(f"❌ No se encontró la carpeta: {ruta_fotos}")
        return
    
    # Escanear todas las subcarpetas
    for item in sorted(os.listdir(ruta_fotos)):
        ruta_categoria = ruta_fotos / item
        
        # Solo procesar si es carpeta
        if os.path.isdir(ruta_categoria):
            archivos = obtener_archivos_fotos(str(ruta_categoria))
            
            # Solo agregar categorías que tengan fotos
            if archivos:
                categorias_activas.append({
                    "cat": item,
                    "archivos": archivos
                })
                print(f"✅ {item}: {len(archivos)} imágenes")
    
    if not categorias_activas:
        print("⚠️ No se encontraron imágenes en las carpetas")
        return
    
    # Generar el archivo JavaScript
    contenido = f"""// Auto-generado por generate-inventory.py
// NO EDITAR MANUALMENTE - ejecuta: python3 generate-inventory.py

const CATEGORIAS_ACTIVAS = {json.dumps(categorias_activas, indent=4, ensure_ascii=False)};

console.log('✅ Inventario cargado:', CATEGORIAS_ACTIVAS.length, 'categorías activas');
"""
    
    # Escribir el archivo
    ruta_salida = Path(__file__).parent / 'inventario-auto.js'
    with open(ruta_salida, 'w', encoding='utf-8') as f:
        f.write(contenido)
    
    print(f"\n✅ Archivo generado: {ruta_salida}")
    print(f"📊 Total de categorías: {len(categorias_activas)}")

if __name__ == '__main__':
    generar_inventario()

File: generate-inventory.js
const fs = require('fs');
const path = require('path');

// Ruta a la carpeta de fotografías
const fotoDir = path.join(__dirname, 'fotografias');

// Función para obtener todas las carpetas de categorías
function getFotosInventario() {
    const categorias = [];
    
    // Leer todas las carpetas en /fotografias
    const items = fs.readdirSync(fotoDir, { withFileTypes: true });
    
    items.forEach(item => {
        // Solo procesar si es una carpeta (no archivos como .png o .jpg)
        if (item.isDirectory()) {
            const categoryPath = path.join(fotoDir, item.name);
            
            // Leer archivos en la carpeta de categoría
            const archivos = fs.readdirSync(categoryPath);
            
            // Filtrar solo imágenes (jpg, jpeg, png, gif, webp)
            const imagenes = archivos.filter(file => {
                const ext = path.extname(file).toLowerCase();
                return ['.jpg', '.jpeg', '.png', '.gif', '.webp'].includes(ext);
            });
            
            // Si hay imágenes, agregar la categoría
            if (imagenes.length > 0) {
                // Agrupar por extensión
                const extensions = {};
                imagenes.forEach(file => {
                    const ext = path.extname(file).toLowerCase().substring(1);
                    extensions[ext] = (extensions[ext] || 0) + 1;
                });
                
                categorias.push({
                    cat: item.name,
                    archivos: imagenes.sort((a, b) => {
                        // Ordenar por número dentro del nombre
                        const numA = parseInt(a.match(/\d+/)?.[0] || 0);
                        const numB = parseInt(b.match(/\d+/)?.[0] || 0);
                        return numA - numB;
                    }),
                    extensions: extensions,
                    total: imagenes.length
                });
            }
        }
    });
    
    return categorias;
}

// Generar el código JavaScript para inventario.js
function generarConfiguracion() {
    const inventario = getFotosInventario();
    
    console.log('✅ Carpetas con fotos encontradas:');
    inventario.forEach(cat => {
        console.log(`   📁 ${cat.cat}: ${cat.total} imágenes`);
    });
    
    // Crear el contenido del archivo
    let contenido = `// Auto-generado por generate-inventory.js
// NO EDITAR MANUALMENTE - ejecuta: node generate-inventory.js

const CATEGORIAS_ACTIVAS = ${JSON.stringify(inventario.map(cat => ({ 
    cat: cat.cat,
    archivos: cat.archivos
})), null, 4)};

// Generar inventario desde los archivos encontrados
const inventario = CATEGORIAS_ACTIVAS.map(cat => ({
    cat: cat.cat,
    archivos: cat.archivos
}));

console.log('Inventario cargado:', CATEGORIAS_ACTIVAS.length, 'categorías activas');
`;
    
    // Escribir el archivo de configuración
    const configPath = path.join(__dirname, 'inventario-auto.js');
    fs.writeFileSync(configPath, contenido);
    
    console.log(`\n✅ Archivo generado: inventario-auto.js`);
    console.log(`📝 Actualiza tu HTML para incluir: <script src="inventario-auto.js"><\/script>`);
    
    return inventario;
}

// Ejecutar generación
generarConfiguracion();
